virt-v2v/cold/entrypoint

```diff
@@ -11,24 +11,68 @@ if [ -f "$1" ] ; then
     LIBGUESTFS_PATH="$APPLIANCE/appliance"
 fi
 
-echo "Run virt-v2v with the following input:"
-cat /mnt/v2v/input.xml
-
-virt-v2v -v -x -i libvirtxml -o null --debug-overlays --no-copy --root=first /mnt/v2v/input.xml
-[ $? != 0 ] && exit 1
-
-echo "Conversion successful. Committing all overlays to local disks."
-for OVERLAY in /var/tmp/*.qcow2
-do
-	if ! qemu-img commit -p "$OVERLAY"
-	then
-		echo Failed to commit overlay "$OVERLAY"!
-		echo Unable to complete import!
-		exit 1
-	fi
+echo "Preparing virt-v2v"
+
+export LIBGUESTFS_PATH=/usr/lib64/guestfs/appliance
+
+# This variable is used to build the list of arguments for virt-v2v.
+args=()
+
+# Temporary storage location
+DIR="/var/tmp/v2v"
+mkdir -p $DIR
+args=("${args[@]}"
+    -o local
+    -os "$DIR"
+)
+
+# Generate disk name suffix from disk number. E.g. "c" (as in "sdc") for
+# 3rd disk.
+gen_name() {
+    local chars=({a..z})
+    local cnt="${#chars[@]}"
+    local i=$(($1 % cnt))
+    local rest=$(($1 / cnt))
+    local prefix=""
+    if [ $rest -gt 0 ] ; then
+        prefix="$(gen_name $rest)"
+    fi
+    echo "$prefix${chars[$i-1]}"
+}
+
+# Disks on filesystem storage.
+# e.g.: /mnt/disks/disk0/disk.img -> vmName-sda
+for disk in /mnt/disks/disk[0-9]* ; do
+	num="${disk:15}"
+	ln -s "$disk/disk.img" "$DIR/$V2V_vmName-sd$(gen_name "$((num+1))")"
+done
+# Disks on block storage.
+# e.g.: /dev/block0 -> vmName-sda
+for disk in /dev/block[0-9]* ; do
+	num="${disk:10}"
+	ln -s "$disk" "$DIR/$V2V_vmName-sd$(gen_name "$((num+1))")"
 done
 
-echo "Commit successful. Cleaning up."
-find /var/tmp -name '*.qcow2' -exec rm -f {} \;
+# Store password to file
+echo -n "$V2V_secretKey" > "$DIR/vmware.pw"
+args=("${args[@]}"
+    -ip "$DIR/vmware.pw"
+)
+
+# Use VDDK if present
+if [ -d "/opt/vmware-vix-disklib-distrib" ]; then
+    args=("${args[@]}"
+        -it vddk
+        -io vddk-libdir=/opt/vmware-vix-disklib-distrib
+        -io "vddk-thumbprint=$V2V_thumbprint"
+    )
+fi
 
-exit 0
+echo "Starting virt-v2v"
+set -x
+ls -l "$DIR"
+exec virt-v2v -v -x \
+    -i libvirt \
+    -ic "$V2V_libvirtURL" \
+    "${args[@]}" \
+    "$V2V_vmName"
```

Comment:
@rwmjones and that's the command that is being executed

Reply:
Unclear what is in args and what the `-ic` URL is, but maybe you're not using VDDK?  The non-VDDK method (using HTTPS) is very slow because VMware uses a Tomcat server.  It would be helpful to see the actual virt-v2v command line.

Reply:
@nyoxi shared the logs offline